osprey-mock-service.js

```diff
@@ -82,16 +96,18 @@ function handler (method) {
   var statusCode = getStatusCode(method)
   var response = (method.responses || {})[statusCode] || {}
   var bodies = response.body || {}
-  var headers = response.headers || method.headers || {}
+  var headers = {}
   var types = Object.keys(bodies)
 
   // Set up the default response headers.
-  Object.keys(headers).forEach(function (key) {
-    var value = headers[key]
-    if (value && value.default) {
-      headers[key] = value.default
-    }
-  })
+  if (Array.isArray(response.headers)) {
+    response.headers.forEach(function (header) {
+      if (header && (header.default || header.example || header.examples)) {
+        var example = getExampleValueFromHeader(header)
+        headers[header.name] = header.default || example
```

Comment:
I would simplify this a bit while making it more explicit (and get rid of `getExampleValueFromHeader()`).
```
if (header.default) {
  headers[header.name] = header.default
} else if (header.example) {
  headers[header.name] = header.example
} else if (header.examples) {
  var randomIndex = Math.floor(Math.random() * header.examples.length)
  headers[header.name] = header.examples[randomIndex].value
}
```
we can also switch the last and second to last statements if `examples` is supposed to supersede `example`.